File: app/src/main/java/com/yonder/exercise/ui/books/IBooksRepository.java
package com.yonder.exercise.ui.books;

import android.arch.lifecycle.LiveData;

import com.yonder.exercise.db.BookModel;

import java.util.List;

/**
 * Created by YusufMac on 02/06/17.
 */

public interface IBooksRepository {

    LiveData<List<BookModel>> getAllBooks();

    void loadBooks(String query);

    void deleteAllBooks();

    void insertAllBooks(List<BookModel> bookModels);
}

File: app/src/main/java/com/yonder/exercise/ui/books/IBooksRepository.kt
package com.yonder.exercise.ui.books

import android.arch.lifecycle.LiveData

import com.yonder.exercise.db.BookModel

/**
 * Created by YusufMac on 08/10/2017.
 */


interface IBooksRepository {

    val allBooks: LiveData<List<BookModel>>
    fun loadBooks(query: String)
    fun deleteAllBooks()
    fun insertAllBooks(bookModels: List<BookModel>)
}